### PR TITLE
P/brentpicasso/pit stop timer config

### LINKED ISSUE #958: Pit stop timer: Make timer increment based on time delta from start, not tick
Due to varying speeds of tablets, cannot count on ticks happening at a predictable interval. 

Change to show the time as a delta from when the timer started.

### PR DESCRIPTION
**Pull request by brentpicasso**
* Make pit stop timer configurable
* Introduce a preferences screen for the dashboard mode, include pit sop timer functions as a first set of preferences
* Correct timer incrementing issue (base it on actual time elapsed, not count on actual ticks happening on time)
* IconButton can now fade in and out as needed
* other refactorings

Resolves
 #958 
#948

### CODE REVIEW

**Comment by brentpicasso** [comment]
setTickRate(10)

sid = addChannel('Speed', 10, 0)
clid = addChannel('CurrentLap', 10, 0)
setChannel(clid, 1)

function onTick() 
local speed = getImu(0) * 100
if speed < 0 then speed = 0 end
setChannel(sid, speed)

end

File: autosportlabs/racecapture/databus/databus.py [hunk 1]
```diff
@@ -146,23 +146,21 @@ def remove_sample_listener(self, listener):
         :param listener
         :type object / callback function
         '''
-        Logger.info('DataBus: remove sample listener {}'.format(listener))
         try:
             self.sample_listeners.remove(listener)
         except Exception as e:
-            Logger.error('Could not remove sample listener {}: {}'.format(listener, e))
```
Comment (stieg) [inline autosportlabs/racecapture/databus/databus.py]: Why the demotion here?  Is this not an error?

File: autosportlabs/racecapture/settings/prefs.py [hunk 2]
```diff
@@ -101,6 +168,13 @@ def set_config_defaults(self):
         self.config.setdefault('preferences', 'last_dash_screen', 'gaugeView')
         self.config.setdefault('preferences', 'global_help', True)
 
+        # Dashboard preferences
+        self.config.adddefaultsection('dashboard_preferences')
+        self.config.setdefault('dashboard_preferences', 'pitstoptimer_enabled', 1)
```
Comment (stieg) [inline autosportlabs/racecapture/settings/prefs.py]: Not sure what you do about magic numbers in Python.  I know in C we put them in `#define` declarations at the top.

File: autosportlabs/racecapture/settings/prefs.py [hunk 3]
```diff
@@ -125,17 +199,91 @@ def load(self):
         except Exception:
             pass
 
+    def get_pref_bool(self, section, option, default=None):
```
Comment (stieg) [inline autosportlabs/racecapture/settings/prefs.py]: Get pref bool default is `None`?  Should that be False?  Or `None` to indicate that the user has no preference?

File: resource/settings/dashboard_settings.json [hunk 4]
```diff
@@ -0,0 +1,35 @@
+[
```
Comment (stieg) [inline resource/settings/dashboard_settings.json]: The settings are a list of objects?  Seems weird?

File: autosportlabs/racecapture/settings/prefs.py (hunk 3 above)
Comment (brentpicasso) [inline autosportlabs/racecapture/settings/prefs.py]: the latter is correct. "Null"

**Comment by stieg** [comment]
Well needs some conflict resolution before I can give this the go-ahead.  Merge in the latest branch and I'll peek again.

File: resource/settings/dashboard_settings.json (hunk 4 above)
Comment (brentpicasso) [inline resource/settings/dashboard_settings.json]: That's how Kivy specifies it.

File: autosportlabs/racecapture/databus/databus.py (hunk 1 above)
Comment (brentpicasso) [inline autosportlabs/racecapture/databus/databus.py]: It's not considered really an error, it was there for debug purposes before.

**Comment by brentpicasso** [comment]
Conflict resolved, thanks.

File: autosportlabs/racecapture/settings/prefs.py (hunk 2 above)
Comment (brentpicasso) [inline autosportlabs/racecapture/settings/prefs.py]: This needs higher level refactoring so defaults are defined in a central place. I think doing that is out of scope for this PR.

File: autosportlabs/racecapture/databus/databus.py (hunk 1 above)
Comment (stieg) [inline autosportlabs/racecapture/databus/databus.py]: Ok.

File: autosportlabs/racecapture/settings/prefs.py (hunk 2 above)
Comment (stieg) [inline autosportlabs/racecapture/settings/prefs.py]: Agreed that it is out of scope.  Do you have a ticket tracking that work?  Might be best to leave a comment referencing the ticket that will get it fixed so its easier to find later.  Sound good?

**Comment by stieg** [comment]
Looks like a clean merge.  :shipit: pending ticket for constants and associated notes.

File: autosportlabs/racecapture/settings/prefs.py (hunk 2 above)
Comment (brentpicasso) [inline autosportlabs/racecapture/settings/prefs.py]: Tracked in #964
